pkg/utils/k8sutil.go

```diff
@@ -730,20 +731,31 @@ func DeployMsgBroker(client *kubernetes.Clientset, kafkaVer string, ctlNamespace
 	return nil
 }
 
-// GetPodName returns pod to which the function is deployed
-func GetPodName(c *kubernetes.Clientset, ns, funcName string) (string, error) {
-	po, err := c.Core().Pods(ns).List(metav1.ListOptions{})
+// GetPods returns list of pods to which the function is deployed
+func GetPods(c *kubernetes.Clientset, ns, funcName string) ([]v1.Pod, error) {
+	funcPods := []v1.Pod{}
+	pods, err := c.Core().Pods(ns).List(metav1.ListOptions{})
 	if err != nil {
-		return "", err
+		return funcPods, err
 	}
 
-	for _, item := range po.Items {
-		if strings.Index(item.Name, funcName) == 0 {
-			return item.Name, nil
+	// finding the pod with label[function]=funcName
+	for _, item := range pods.Items {
+		if item.ObjectMeta.Labels["function"] == funcName {
+			funcPods = append(funcPods, item)
 		}
 	}
+	return funcPods, nil
+}
 
-	return "", errors.New("Can't find pod starting with the function name")
+// GetReadyPod returns the first pod has passed the liveness probe check
+func GetReadyPod(pods []v1.Pod) v1.Pod {
+	for _, pod := range pods {
+		if pod.Status.ContainerStatuses[0].Ready {
+			return pod
+		}
+	}
+	return v1.Pod{}
```

Comment:
Hrm, perhaps return a pointer to the list element, and return `nil` in the "none ready" case?

Or better, return a `pod, ok` tuple that will encourage callers to actually check the result.  Still return a reference just to avoid copying the Pod structure.